### PR TITLE
[WIP] prototype of redirectable DisplayList geometry definitions

### PR DESCRIPTION
**Pull request by flar**
This is a prototype of a mechanism that will allow us to move the engine from Skia geometry objects to Impeller geometry objects with minimal disturbance for engine developers. This PR will not be pushed as is, but the PR is being created to check how the mechanism passes CI tests. Eventually the change will happen in up to 4 stages:

1. Using Skia via C++ `using` aliases (the intended initial state)
2. Using Skia via adapter structs/classes (the default state in this PR, but only for early testing of the concept)
3. Using Impeller via adapter structs/classes (currently prototyped, but non-functional)
4. Using Impeller via C++ `using` aliases (this is the intended end result, TBD when we get here)

The reason for these 4 stages is to avoid a situation where the switch to Impeller requires a massive PR that changes nearly every file in the engine which might need to be reverted/relanded a few times to get it right. This could be done in as little as 2 stages if the middle pair of stages turns out to not be feasible, but there will be a big spanning change as a result. The 4 stages keep every change minimal with a high probability of being "semantic only".

### TLDR

- Step 1 allows us to do renames throughout the engine without any compatibility issues.
- Steps 2 and 3 allow us to migrate a fully renamed engine from "thin Skia adapters, heavy Impeller adapters" to "heavy Skia adapters, thin Impeller adapters" over time with minimal piece-wise edits to the engine code
- Step 4 leaves us with an engine running on Impeller with no code changes other than renames.

### Step 1: Quick, Easy, Repetitive

Starting in Stage 1 we've create a series of aliases for Skia objects in one initial PR and then follow it with a number of conversion PRs that slowly change the naming used within the engine to the new names that will eventually point to Impeller objects. There should be no semantic change during this phase, only a nomenclature change, so no changes in performance or behavior should happen unless a name replacement gets "typoed". We can also convert the code in parallel smaller PRs since there are no changes visible to the compiler - `using` aliases are considered identical.

### Step 2: Very Quick, Very Easy, Sets the Stage

Once the entire engine is renamed, we can switch to the Skia adapter classes which will provide a basis for the conversion. The adapters are really just thin renames of the classes themselves using inheritance. We need the adapters to implement the switch to Impeller because there will be methods that need to have their names changed, minimally, and possibly their behavior changed (i.e. Skia uses degrees where Impeller might use radians, etc.) This stage should be very quick and is essentially already prepared in this prototype, but it can't be turned on until the naming changes in Stage 1 are completed as all components must be using the new names before the adapters can work(*). It should not change any semantics, but it might include some implicit copy constructors where we internally receive an `SkFoo` object as the return of an internal Skia method and that needs to be wrapped with the equivalent `DlFoo` object (I'm looking at you, factory constructors). This conversion should be the only change here so the switch should be quick, have no behavioral changes, and we'll just need to monitor performance changes to determine if we want this stage to be shipped or just remain internal for testing.

We don't actually need to ship the engine in Stage 2, we can continue to ship Stage 1 and use Stage 2 internally for testing and development of Stage 3. Stage 3 will be to implement the Impeller adapters which will mimic Skia APIs on top of Impeller objects.

### Step 3: Engine Can Finally Run on Impeller (though through temporary adapters)

Next we simply mirror the Skia API method names and semantics in the Impeller adapter struct/classes so that we can run on top of them and check functionality. Performance might be acceptable in this state if most of the adapter methods are simple renames and the compiler is smart enough to inline them. We may or may not ship the engine on top of the Impeller adapters (just as we may or may not ship under Stage 2), we'll have to see how functional it is when we get there. But it will set us up for the final step.

### Step 4: Lots of Little Changes To Get From SkHere to impeller::There

The biggest and most important step will be to convert areas of the engine from Skia method names/semantics to Impeller method names/semantics which will move the burden of "adapter" work from the Impeller adapters to the Skia adapters. This can be done piecemeal in a number of smaller PRs to minimize their footprint target area for regressions, but only if the performance is reasonable during the process. During and when this is done, the Impeller adapters will slowly become token adapters similar to the starting state for the Skia adapters and at that stage the Skia adapters will be the complicated ones.

### Step 5: The fun part

The final step will be to switch to Impeller aliases and then delete all of the adapters and switch code. Note that the final state includes keeping the `using DlFoo = impeller::Foo` aliases to avoid "prefix bloat" in the engine code.

### The Part Where I Sorta Lied Above

(*) technically we can change a few APIs here and there using the adapters as I've done in this initial proof of concept PR since there is automatic conversion from Sk-object to Fl-object, but we end up with corner cases like `vector<SkFoo>` cannot be communicated to someone expecting `vector<DlFoo>` and vice versa. That works between converted and un-converted code when we have `using` aliases, but not with inheriting adapter sub-classes. This reason is why this "proof of concept" PR touches so many files: one file converted to using DlRect and running on the adapters caused errors throughout the engine that had to be fixed before it could compile. The errors in the Fuchsia code in this PR indicate the dangers of starting out with adapter classes and converting only "most" of the code.

### CODE REVIEW

**Comment by flar** [comment]
The first place to start looking at (not really reviewing as this is not intended to land) would be the code in `display_list/geometry/dl_geometry_types.h` and its associated skia/impeller helper includes.

**Comment by flar** [comment]
Proof of concept achieved, but this work is not a priority at this time.
